Source/Orts.Simulation/Simulation/Signalling/Signals.cs

```diff
@@ -10384,6 +10361,39 @@ public TrackMonitorSignalAspect TranslateTMAspect(MstsSignalAspect SigState)
 
             return (newRoute);
         }
+
+        //================================================================================================//
+        /// <summary>
+        /// number of remaining signals to clear
+        /// </summary>
+
+        public int GetReqNumClearAheadExplorer(bool isPropagated, int signalNumClearAhead)
+        {
+            int ReqNumClearAhead = 0;
+
+            if (SignalNumClearAhead_MSTS > -2)
+            {
+                ReqNumClearAhead = propagated ?
+                    signalNumClearAhead - SignalNumNormalHeads : SignalNumClearAhead_MSTS - SignalNumNormalHeads;
+            }
+            else
+            {
+                if (SignalNumClearAheadActive == -1)
+                {
+                    ReqNumClearAhead = propagated ? signalNumClearAhead : 1;
+                }
+                else if (SignalNumClearAheadActive == 0)
+                {
+                    ReqNumClearAhead = 0;
+                }
+                else
+                {
+                    ReqNumClearAhead = isPropagated ? signalNumClearAhead - 1 : SignalNumClearAheadActive - 1;
+                }
+            }
+
+            return ReqNumClearAhead;
```

Comment:
Can be simplified to
```suggestion
            if (SignalNumClearAhead_MSTS > -2)
                return propagated ? signalNumClearAhead - SignalNumNormalHeads : SignalNumClearAhead_MSTS - SignalNumNormalHeads;
            else if (SignalNumClearAheadActive == -1)
                return propagated ? signalNumClearAhead : 1;
            else if (SignalNumClearAheadActive == 0)
                return 0;
            else
                return isPropagated ? signalNumClearAhead - 1 : SignalNumClearAheadActive - 1;
```

Reply:
This suggestion shows the difference between software developers and software testers.
Developers like compressed code, but testers prefer singular flows. I've spend most of my professional life as software test engineer, and in that time I build up a firm dislike of multiple-exit methods as it often makes debugging more complex.

Reply:
Another point about this change.
This logic is now part of method TranslateTMAspect. This method was created to translate the signal enumeration to the cabview enumeration, and as such it has no function within the signal logic.
Admittedly I do not know if the code w.r.t. SNCA was introduced as part of this commit or if it was introduced earlier, for I have not kept up with the changes made to OR for quite some time now.
But this logic has no place in this method. As it has now been collected in a separate method, I would suggest to remove the call to this method from TranslateTMAspect and move is to the location where TranslateTMAspect is called, and keep TranslateTMAspect clean as translation between enums only.

Reply:
I think it's just a 'git diff' error. This code was in function requestClearSignalExplorer(), but I moved it to a new function GetReqNumClearAheadExplorer(), in order to use this from CheckExplorerPath() too. I haven't added SNCA to TranslateTMAspect, of course.

Reply:
Starting the function with `int ReqNumClearAhead = 0;` actually suppresses the type checker. It will no longer warn you when you've forgotten to assign a value to it in one of the if branches.